stl/inc/atomic

```diff
@@ -131,6 +131,68 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 #define ATOMIC_LLONG_LOCK_FREE    2
 #define ATOMIC_POINTER_LOCK_FREE  2
 
+#if _HAS_CXX20
+extern "C" {
+void __stdcall __std_atomic_wait_direct(const void* const _Storage, const void* const _Comparand,
+    const std::size_t _Size, unsigned long long& _Wait_context) noexcept;
+void __stdcall __std_atomic_notify_one_direct(const void* const _Storage) noexcept;
+void __stdcall __std_atomic_notify_all_direct(const void* const _Storage) noexcept;
+
+void __stdcall __std_atomic_unwait_direct(const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+
+void __stdcall __std_atomic_wait_indirect(const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+void __stdcall __std_atomic_notify_one_indirect(const void* const _Storage) noexcept;
+void __stdcall __std_atomic_notify_all_indirect(const void* const _Storage) noexcept;
+
+void __stdcall __std_atomic_unwait_indirect(const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+
+std::size_t __stdcall __std_atomic_get_spin_count(const bool _Is_direct) noexcept;
+
+enum _Atomic_spin_phase : unsigned long long {
+    _Atomic_wait_phase_mask            = 0x0000'000F,
+    _Atomic_spin_value_mask            = 0xFFFF'FFF0,
+    _Atomic_spin_value_step            = _Atomic_wait_phase_mask + 1,
+    _Atomic_counter_value_mask         = ~_Atomic_wait_phase_mask,
+    _Atomic_counter_value_step         = _Atomic_wait_phase_mask + 1,
+    _Atomic_wait_phase_init_spin_count = 0x0000'0000,
+    _Atomic_wait_phase_spin            = 0x0000'0008,
+    _Atomic_wait_phase_wait_locked     = 0x0000'0001,
+    _Atomic_wait_phase_wait_none       = 0x0000'0002,
+    _Atomic_wait_phase_wait_counter    = 0x0000'0004,
+    _Atomic_unwait_needed              = _Atomic_wait_phase_wait_locked,
+};
+}
+
+inline bool __std_atomic_wait_spin(unsigned long long& _Wait_context, const bool _Is_direct) noexcept {
+#ifdef _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+    if (_Is_direct) {
+        return false;
+    } else {
+#endif // _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+        switch (_Wait_context & _Atomic_wait_phase_mask) {
+        case _Atomic_wait_phase_init_spin_count: {
+            _Wait_context = _Atomic_wait_phase_spin | __std_atomic_get_spin_count(_Is_direct);
+            [[fallthrough]];
+        }

```

Comment:
I suppose `switch` is an exception to our convention of "all control flow must be braced, no exceptions". We don't add braces to cases except when they define variables. Occurs below.

(In a switch, the control flow is naturally divided by `case` and `default` so I don't believe that braces add any value - I could be persuaded otherwise though.)